br-webui/server1.js

```diff
@@ -0,0 +1,191 @@
+const child_process = require('child_process');
+const dgram = require('dgram');
+const SocketIOFile = require('socket.io-file');
+var logger = require('tracer').console();
+var os = require("os");
+var env = process.env
+const multer = require('multer');					//file storing middleware
+const bodyParser = require('body-parser'); 
+logger.log('ENVIRONMENT', process.env)
+logger.log('COMPANION_DIR', process.env.COMPANION_DIR)
+app.use(express.static('public'));
+app.use('/webui.log', express.static('/home/pi/.webui.log'));
+app.use('/js', express.static(__dirname + '/node_modules/bootstrap/dist/js')); // redirect bootstrap JS
+app.use('/js', express.static(__dirname + '/node_modules/jquery/dist')); // redirect JS jQuery
+app.use('/font-awesome', express.static(__dirname + '/node_modules/font-awesome')); // redirect JS jQuery
+app.use('/css', express.static(__dirname + '/node_modules/bootstrap/dist/css')); // redirect CSS bootstrap
+app.use('/style.css', express.static(__dirname + '/style.css')); // redirect CSS bootstrap
+app.use('/js', express.static(__dirname + '/node_modules/network-js/dist'));
+app.use('/js', express.static(__dirname + '/node_modules/bootstrap-switch/dist/js'));
+app.use('/css', express.static(__dirname + '/node_modules/bootstrap-switch/dist/css/bootstrap2'));
+app.use('/js', express.static(__dirname + '/node_modules/bootstrap-select/dist/js'));
+app.use('/css', express.static(__dirname + '/node_modules/bootstrap-select/dist/css'));
+app.use('/js', express.static(__dirname + '/node_modules/bootstrap-slider/dist'));
+app.use('/css', express.static(__dirname + '/node_modules/bootstrap-slider/dist/css'));
+
+app.use (bodyParser.urlencoded({extended:false})); 	//handle body requests
+app.use(bodyParser.json())
+var fs = require("fs");
+var expressLiquid = require('express-liquid');
+var options = {
+	//read file handler, optional 
+	includeFile: function (filename, callback) {
+		fs.readFile(filename, 'utf8', callback);
+	},
+	// the base context, optional 
+	context: expressLiquid.newContext(),
+	// custom tags parser, optional 
+	customTags: {},
+	// if an error occurred while rendering, show detail or not, default to false 
+	traceError: false
+};
+app.set('view engine', 'liquid');
+app.engine('liquid', expressLiquid(options));
+app.use(expressLiquid.middleware);
+
+// Companion repository root directory
+var _companion_directory = process.env.COMPANION_DIR;
+
+app.get('/', function(req, res) {
+	res.render('index', {});
+});
+/*app.get('/socket.io-file-client.js', (req, res, next) => {
+	return res.sendFile(__dirname + '/node_modules/socket.io-file-client/socket.io-file-client.js');
+});*/
+
+var server = app.listen(5000, function() {
+	var host = server.address().address;
+	var port = server.address().port;
+	logger.log("App running at http://%s:%s", host, port);
+});
+
+const multerConfig = {
+
+	storage: multer.diskStorage({
+		destination: function(req,file,next) {
+			next(null, '/mnt/mainfs/home/pi/new_files');
```

Comment:
are you going to store the files in the mainfs (not recovery) ? How are you going to read the .img file if this partition will be formatted ?

Reply:
the recovery partition does not have enough space to save a 1.4 gb file. thats why im storing it on the other partition where the update is supposed to happen. and once its saved we can just extract the image over existing partition using dd. so there will be no issues with accessing it.